pkg/handlers/generic/mutation/controlplanevirtualip/inject.go

```diff
@@ -187,3 +167,36 @@ func (h *ControlPlaneVirtualIP) Mutate(
 		},
 	)
 }
+
+func deleteFiles(obj *controlplanev1.KubeadmControlPlaneTemplate, filePathsToDelete ...string) {
+	for i := len(obj.Spec.Template.Spec.KubeadmConfigSpec.Files) - 1; i >= 0; i-- {
+		for _, path := range filePathsToDelete {
+			if obj.Spec.Template.Spec.KubeadmConfigSpec.Files[i].Path == path {
+				obj.Spec.Template.Spec.KubeadmConfigSpec.Files = slices.Delete(
+					obj.Spec.Template.Spec.KubeadmConfigSpec.Files, i, i+1,
+				)
+				break
+			}
+		}
+	}
+}
+
+// mergeFiles will merge the files into the KubeadmControlPlaneTemplate,
+// overriding any file with the same path and appending the rest.
+func mergeFiles(obj *controlplanev1.KubeadmControlPlaneTemplate, filesToMerge ...bootstrapv1.File) {
```

Comment:
nit. same as above.

```
func mergeFiles(files []bootstrapv1.File, filesToMerge ...bootstrapv1.File) []bootstrapv1.File
```

Reply:
Good idea, done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1062/commits/49381076065595dd41dc82dc169e480170ab9ca1